File: src/main/java/com/teambr/bookshelf/common/container/ICustomSlot.java
package com.teambr.bookshelf.common.container;

import com.teambr.bookshelf.collections.Couplet;

public interface ICustomSlot {
    enum SLOT_SIZE {
        STANDARD,
        LARGE
    }

    SLOT_SIZE getSlotSize();

    Couplet<Integer, Integer> getPoint();
}

File: src/main/scala/com/teambr/bookshelf/common/container/ICustomSlot.scala
package com.teambr.bookshelf.common.container

/**
 * This file was created for Bookshelf
 *
 * Bookshelf is licensed under the
 * Creative Commons Attribution-NonCommercial-ShareAlike 4.0 International License:
 * http://creativecommons.org/licenses/by-nc-sa/4.0/
 *
 * @author Paul Davis <pauljoda>
 * @since August 04, 2015
 */
abstract class ICustomSlot {
    def getSlotSize: SLOT_SIZE.Value

    def getPoint: (Integer, Integer)
}

object SLOT_SIZE extends Enumeration {
    val STANDARD, LARGE = Value
}
